control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -782,6 +795,52 @@ func (c *GatewayController) getRelatedHTTPRoutes(ctx context.Context, gateway ty
 	return list.Items, nil
 }
 
+func (c *GatewayController) getExternalFiltersForHTTPRoute(ctx context.Context, route gwv1beta1.HTTPRoute, resources *common.ResourceMap) ([]interface{}, error) {
+	var externalFilters []interface{}
+	for _, rule := range route.Spec.Rules {
+		for _, filter := range rule.Filters {
+			c.Log.Info("\n\n filter iterating")
+			c.Log.Info(string(filter.Type))
+			var externalFilter client.Object
+			//check to see if we need to grab this filter
+			if filter.ExtensionRef == nil {
+				continue
+			}
+			switch kind := filter.ExtensionRef.Kind; kind {
+			case "HTTPRouteRetryFilter":
+				externalFilter = &v1alpha1.HTTPRouteRetryFilter{}
+			case "HTTPRouteTimeoutFilter":
+				externalFilter = &v1alpha1.HTTPRouteTimeoutFilter{}
+			default:
+				externalFilter = &common.HTTPRouteFilterError{
+					Kind:    common.ErrInvalidExternalRefType,
+					Message: fmt.Sprintf("%s not a valid externalref for httproutefilter", kind),
+				}
+				resources.ReferenceCountHTTPRouteFilter(route, externalFilter)
+				continue
+			}
+
+			//get object from API
+			if err := c.Client.Get(ctx, client.ObjectKey{
+				Name:      string(filter.ExtensionRef.Name),
+				Namespace: route.Namespace,
+			}, externalFilter); err != nil {
+				c.Log.Info("\n\n\n error")
+				c.Log.Error(err, "unable to get filter")
+				externalFilter = &common.HTTPRouteFilterError{
+					Kind:    common.ErrExternalRefNotFound,
+					Message: err.Error(),
+				}
+			}
```

Comment:
#nit this is the kind of place where I avoid the compound `if` because readability suffers

```suggestion
			err := c.Client.Get(ctx, client.ObjectKey{
				Name:      string(filter.ExtensionRef.Name),
				Namespace: route.Namespace,
			}, externalFilter)
			if err != nil {
				c.Log.Info("\n\n\n error")
				c.Log.Error(err, "unable to get filter")
				externalFilter = &common.HTTPRouteFilterError{
					Kind:    common.ErrExternalRefNotFound,
					Message: err.Error(),
				}
			}
```